### PR TITLE
Fixes #13666: Jinja templating method does not correctly clean up temporary files in audit mode

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/13666

### CODE REVIEW

File: tree/30_generic_methods/file_from_template_jinja2.cf [hunk 1]
```diff
@@ -243,8 +243,10 @@ bundle agent file_from_template_jinja2(source_template, destination)
       "copy classes" usebundle => _classes_copy("file_copy_from_local_source_${destination_canon}", "${old_class_prefix}");
 
     pass3.state_dumped::
+      "remove_dry_run_mode" usebundle => push_dry_run_mode("false");
```
Comment (fanf) [inline tree/30_generic_methods/file_from_template_jinja2.cf]: Could you had a comment to explain why we need to override audit mode here?

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
OK, merging this PR

**Comment by Fdall** [comment]
OK, merging this PR

**Pull request merged**